backends/aoti/runtime/AotiBackend.cpp

```diff
@@ -0,0 +1,532 @@
+// They can be backed by ET systems
+
+using namespace std;
+
+using executorch::aten::ScalarType;
+using executorch::runtime::ArrayRef;
+using executorch::runtime::Backend;
+using executorch::runtime::BackendExecutionContext;
+using executorch::runtime::BackendInitContext;
+using executorch::runtime::CompileSpec;
+using executorch::runtime::DelegateHandle;
+using executorch::runtime::Error;
+using executorch::runtime::EValue;
+using executorch::runtime::FreeableBuffer;
+using executorch::runtime::MemoryAllocator;
+using executorch::runtime::Result;
+using executorch::runtime::etensor::Tensor;
+
+
+extern "C" {
+  using AOTITensorHandle = Tensor*;
+
+  // TODO: We should get a proper one
+  struct CUDAStreamGuardOpaque;
+  using CUDAStreamGuardHandle = CUDAStreamGuardOpaque*;
+
+  using AOTIRuntimeError = Error;
+  using AOTITorchError = Error;
+
+  struct AOTInductorModelContainerOpaque;
+  using AOTInductorModelContainerHandle = AOTInductorModelContainerOpaque*;
+  using AOTInductorStreamHandle = void*;
+  using AOTIProxyExecutorHandle = void*;
+
+  using AOTInductorModelContainerCreateWithDeviceFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle* container_handle,
+    size_t num_models,
+    const char* device_str,
+    const char* cubin_dir);
+
+  using AOTInductorModelContainerDeleteFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle container_handle);
+
+  using AOTInductorModelContainerGetNumInputsFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle container_handle,
+    size_t* num_constants);
+
+  using AOTInductorModelContainerGetNumOutputsFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle container_handle,
+    size_t* num_constants);
+
+  using AOTInductorModelContainerRunFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle container_handle,
+    AOTITensorHandle* input_handles, // array of input AOTITensorHandle; handles
+                                     // are stolen; the array itself is borrowed
+    size_t num_inputs,
+    AOTITensorHandle*
+        output_handles, // array for writing output AOTITensorHandle; handles
+                        // will be stolen by the caller; the array itself is
+                        // borrowed
+    size_t num_outputs,
+    AOTInductorStreamHandle stream_handle,
+    AOTIProxyExecutorHandle proxy_executor_handle);
+
+  AOTInductorModelContainerCreateWithDeviceFunc AOTInductorModelContainerCreateWithDevice = nullptr;
+  AOTInductorModelContainerDeleteFunc AOTInductorModelContainerDelete = nullptr;
+  AOTInductorModelContainerGetNumInputsFunc AOTInductorModelContainerGetNumInputs = nullptr;
+  AOTInductorModelContainerGetNumOutputsFunc AOTInductorModelContainerGetNumOutputs = nullptr;
+  AOTInductorModelContainerRunFunc AOTInductorModelContainerRun = nullptr;
+  std::unordered_map<Tensor*, std::vector<int64_t>> tensor_to_sizes;
+  std::unordered_map<Tensor*, std::vector<int64_t>> tensor_to_strides;
```

Comment:
ideally these should be DimVector, not std::vector. open problem: ExecuTorch doesn't have DimVector or c10::SmallVector and can't easily get at them because 1) this shouldn't link libtorch 2) SmallVector has and should have a .cpp file.